### PR TITLE
[chore] update k8s.io/client-go dependency

### LINKED ISSUE #21032: [chore] Upgrade k8s.io/client-go dependency
k8s.io/client-go 0.27.0 has breaking changes in `ResourceEventHandler` interface introduced in https://github.com/kubernetes/client-go/commit/5d70a118df80fd8ddb8fbd041af95acae49ae617

We have a few implementations of the interface that are causing the upgrade to fail:

```
Error: ../../extension/observer/k8sobserver/handler.go:28:36: cannot use (*handler)(nil) (value of type *handler) as type "k8s.io/client-go/tools/cache".ResourceEventHandler in variable declaration:
	*handler does not implement "k8s.io/client-go/tools/cache".ResourceEventHandler (wrong type for OnAdd method)
		have OnAdd(objectInterface interface{})
		want OnAdd(obj interface{}, isInInitialList bool)
Error: ../../extension/observer/k8sobserver/extension.go:61:45: cannot use k.handler (variable of type *handler) as type "k8s.io/client-go/tools/cache".ResourceEventHandler in argument to podInformer.AddEventHandler:
	*handler does not implement "k8s.io/client-go/tools/cache".ResourceEventHandler (wrong type for OnAdd method)
		have OnAdd(objectInterface interface{})
		want OnAdd(obj interface{}, isInInitialList bool)
Error: ../../extension/observer/k8sobserver/extension.go:70:46: cannot use k.handler (variable of type *handler) as type "k8s.io/client-go/tools/cache".ResourceEventHandler in argument to nodeInformer.AddEventHandler:
	*handler does not implement "k8s.io/client-go/tools/cache".ResourceEventHandler (wrong type for OnAdd method)
		have OnAdd(objectInterface interface{})
		want OnAdd(obj interface{}, isInInitialList bool)
# github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor/internal/kube
Error: ../../processor/k8sattributesprocessor/internal/kube/fake_informer.go:54:9: cannot use handler (variable of type "k8s.io/client-go/tools/cache".ResourceEventHandler) as type "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration in return statement:
	"k8s.io/client-go/tools/cache".ResourceEventHandler does not implement "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration (missing HasSynced method)
Error: ../../processor/k8sattributesprocessor/internal/kube/fake_informer.go:148:9: cannot use handler (variable of type "k8s.io/client-go/tools/cache".ResourceEventHandler) as type "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration in return statement:
	"k8s.io/client-go/tools/cache".ResourceEventHandler does not implement "k8s.io/client-go/tools/cache".ResourceEventHandlerRegistration (missing HasSynced method)
```

Need to update the code to use the new interfaces and upgrade the k8s go library

### PR DESCRIPTION
**Pull request by bryan-aguilar**
**Description:** Upgrade `k8s.io/client-go` dependency in `k8sattributesprocessor` and `k8sobserver`. This update required code changes due to breaking interfaces changes in the dependency. 

**Link to tracking Issue:** closes #21032 

**Testing:** unit test

**Documentation:** n/a

### CODE REVIEW

**Comment by bryan-aguilar** [comment]
Can this get a `skip changelog` label?

File: processor/k8sattributesprocessor/internal/kube/fake_informer.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ func (f *NoOpInformer) AddEventHandler(handler cache.ResourceEventHandler) (cach
 	return f.AddEventHandlerWithResyncPeriod(handler, time.Second)
 }
 func (f *NoOpInformer) AddEventHandlerWithResyncPeriod(handler cache.ResourceEventHandler, resyncPeriod time.Duration) (cache.ResourceEventHandlerRegistration, error) {
-	return handler, nil
+	return nil, nil
```
Comment (atoulme) [inline processor/k8sattributesprocessor/internal/kube/fake_informer.go:148]: Is this change intended?

**Comment by atoulme** [comment]
> Can this get a `skip changelog` label?

No need. You have `[chore]` in the title of your PR, which does the same thing.

File: processor/k8sattributesprocessor/internal/kube/fake_informer.go (hunk 1 above)
Comment (bryan-aguilar) [reply processor/k8sattributesprocessor/internal/kube/fake_informer.go:148]: Yes, this is a fake informer that is only used for testing from what I can tell. This was created to satisfy the `[SharedInformer](https://pkg.go.dev/k8s.io/client-go/tools/cache@v0.27.1#SharedInformer)` interface

Comment (bryan-aguilar) [reply processor/k8sattributesprocessor/internal/kube/fake_informer.go:148]: I asked myself that question also but could not find any reason why we could not just return nil here. Unit tests pass and I am hoping that e2e test show the same. I'm fixing the go mod issue as we speak.

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
Thanks @bryan-aguilar!

**Pull request merged**